source/OptimaJet.DWKit.StarterApplication/Services/ProjectService.cs

```diff
@@ -0,0 +1,102 @@
+
+        public ProjectService(
+            IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
+            UserRepository userRepository,
+            IEntityRepository<Project> projectRepository,
+            GroupRepository groupRepository,
+            IEntityRepository<Organization> organizationRepository,
+            ILoggerFactory loggerFactory) : base(jsonApiContext, projectRepository, loggerFactory)
+        {
+            OrganizationContext = organizationContext;
+            JsonApiContext = jsonApiContext;
+            CurrentUserContext = currentUserContext;
+            UserRepository = userRepository;
+            GroupRepository = groupRepository;
+            OrganizationRepository = organizationRepository;
+            ProjectRepository = (ProjectRepository)projectRepository;
+        }
+        public override async Task<IEnumerable<Project>> GetAsync()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist)
+            {
+                return Enumerable.Empty<Project>().AsQueryable();
+            }
+            else
+            {
+                var query = JsonApiContext.QuerySet;
+                if (query == null)
+                {
+                    query = new JsonApiDotNetCore.Internal.Query.QuerySet();
+                    JsonApiContext.QuerySet = query;
+                }
+                var value = OrganizationContext.HasOrganization ? OrganizationContext.OrganizationId.ToString() : "";
+                query.Filters.Add(new JsonApiDotNetCore.Internal.Query.FilterQuery("organization-header", value, "="));
+                return await base.GetAsync();
+            }
+        }
+        public override async Task<Project> GetAsync(int id)
+        {
+            var projects = await GetAsync();
+            return projects.SingleOrDefault(g => g.Id == id);
+        }
+        public override async Task<Project> UpdateAsync(int id, Project resource)
+        {
+            //If changing organization, validate the change
+            var updateForm = new Forms.Projects.UpdateForm(UserRepository,
+                                                           GroupRepository,
+                                                           CurrentUserContext,
+                                                           OrganizationRepository,
+                                                           ProjectRepository);
+            if (!updateForm.IsValid(id, resource))
+            {
+                throw new JsonApiException(updateForm.Errors);
+            }
+            // Make sure that group may be accessed by current user
+            var project = await GetAsync(id);
+            if (project == null)
+            {
+                throw new JsonApiException(404, $"Project Id '{id}' not found."); ;
+            }
+            return await base.UpdateAsync(id, resource);
```

Comment:
I wonder if it'd make more sense to rename this `resource` argument to be more contextual to it's use / origin.

maybe: `ModelFromParams` or something.